contracts/schemes/VoteInOrganizationScheme.sol

```diff
@@ -34,21 +34,35 @@ contract VoteInOrganizationScheme is VotingMachineCallbacks, ProposalExecuteInte
     /**
      * @dev initialize
      * @param _avatar the avatar this scheme referring to.
-     * @param _votingMachine the voting machines address to
-     * @param _votingParams genesisProtocol parameters - valid only if _voteParamsHash is zero
-     * @param _voteOnBehalf genesisProtocol parameter - valid only if _voteParamsHash is zero
-     * @param _voteParamsHash voting machine parameters.
+     * @param _votingParams genesisProtocol parameters
+     * @param _voteOnBehalf  parameter
+     * @param _daoFactory  DAOFactory instance to instance a votingMachine.
+     * @param _stakingToken (for GenesisProtocol)
+     * @param _packageVersion packageVersion to instance the votingMachine from.
+     * @param _votingMachineName the votingMachine contract name.
      */
     function initialize(
         Avatar _avatar,
-        IntVoteInterface _votingMachine,
         uint256[11] calldata _votingParams,
         address _voteOnBehalf,
-        bytes32 _voteParamsHash
+        DAOFactory _daoFactory,
+        address _stakingToken,
+        uint64[3] calldata _packageVersion,
```

Comment:
Maybe we could take that from the Avatar or somewhere else? I don't really like how this is duplicated in all the code... (Also relevant for daofactory)

Reply:
the alternative is to always use lasted package for the voting machines.
need to give it more thoughts.